### ISSUE TITLE
📝 Biome doesn't format `.jsonc` file with trailing commas

### ISSUE THREAD

**Issue by Sec-ant** [biomejs/biome#1659]
### Environment information

```bash
❯ npx biome rage
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.6.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "formatter": {
    "enabled": true
  },
  "overrides": [
    {
      "include": ["*.jsonc"],
      "json": {
        "parser": {
          "allowComments": true,
          "allowTrailingCommas": true
        }
      }
    }
  ]
}
```

### Playground link

https://github.com/Sec-ant/biome-jsonc-formatting-bug-repro

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
emmm, I seem to mess up the CSB link. You can test it with a `.jsonc` file like this:
```jsonc
{"foo":          "bar",
}
```

**Comment by Sec-ant** [comment]
Also, FWIW, there's a discussion around the trailing comma formatting behavior of `jsonc` files in Prettier recently: https://github.com/prettier/prettier/issues/15956

I'm curious about what position will Biome take in this question?

**Comment by ematipico** [comment]
Personally, I would like to not deviate too much from the specs. Fortunately, we offer the option to allow trailing commas. That should help. Unless there's a  bug when you enable `parser.json.allowTrailingCommas`

**Comment by Sec-ant** [comment]
> Unless there's a bug when you enable `parser.json.allowTrailingCommas`

Yes that's what this issue is about. But I have trouble sharing the link to my repro in the CSB devbox (the link always jumps to a new template, I don't know why), so I created a github repo here: https://github.com/Sec-ant/biome-jsonc-formatting-bug-repro Thanks for checking this.

**Comment by Sec-ant** [comment]
> Personally, I would like to not deviate too much from the specs. Fortunately, we offer the option to allow trailing commas. That should help.

Ok, let me rephrase my question in a more specific way: If I configure biome in a way that trailing commas are allowed for all the `.jsonc` files, and I have a `.jsonc` file that do not have trailing commas that need to be formatted, will trailing commas be added to the file after formatting?

**Comment by Sec-ant** [comment]
I not very confident with the codebase but I think the switch to support it is here, is this planned to be fixed @ematipico ?

https://github.com/biomejs/biome/blob/f490b15e4542f856d0b8cf8b45c0c7ba51170089/crates/biome_json_formatter/src/separated.rs#L55

**Comment by ematipico** [comment]
Working on this issue. It seems that our tests weren't optimal, and I can see that this isn't working at all.

Sorry for neglecting the issue
